larpandoracontent/LArHelpers/LArMCParticleHelper.h

```diff
@@ -129,6 +153,15 @@ class LArMCParticleHelper
      */
     static const pandora::MCParticle *GetPrimaryMCParticle(const pandora::MCParticle *const pMCParticle);
 
+    /**
+     *  @brief  Get the leasding particle in the hierarchy, for use at ProtoDUNE
+     *
+     *  @param  pMCParticle the input mc particle
+     *
+     *  @return address of the primary parent mc particle
+     */
+    static const pandora::MCParticle *GetLeadingMCParticle(const pandora::MCParticle *const pMCParticle, const int hierarchyTierLimit = 1);
```

Comment:
Is the limit likely to be a parameter of our algorithms or, realistically, always set to 1? (Definitely not saying we should hard code it!!!)

Reply:
I think that it is likely that it will be a parameter of the algorithms because for neutrino interactions we currently drill down to tier 1 (tier 0 being the neutrino and 1 being the primaries, but this is a weird case as we don't see tier 0 particles) and for ProtoDUNE we have been going to tier 0 (the parent test beam particle) and now we want to go to 1.  Ultimately, the next step for this is to go to higher tiers for DUNE (secondary interactions etc), so with that in mind, I wanted to make these changes easily extendible for future use cases, which is why I wanted this to be configurable.